.github/workflows/main.yml

```diff
@@ -51,21 +51,196 @@ jobs:
           cd ..
           # Remove them to avoid confusing pytest.
           rm -rf texar-pytorch
-      - name: Install the project
-        run: |
-          pip install --progress-bar off .[nltk,varder,stanza,elastic,faiss,spacy,allennlp,cliner]
       - name: Lint with flake8
         run: |
           # stop the build if there are Python syntax errors or undefined names
-          flake8 forte_wrapper/ --count --select=E9,F63,F7,F82 --show-source --statistics --exclude=*/cliner/*
+          flake8 forte_wrapper/ --count --select=E9,F63,F7,F82 --show-source --statistics
           # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-          flake8 forte_wrapper/ --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics --exclude=*/cliner/*
+          flake8 forte_wrapper/ --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
       - name: Lint with pylint
         run: |
           pylint forte_wrapper/
       - name: Lint with mypy
         run: |
           mypy .
+
+  test_allennlp:
+    needs: build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ 3.6, 3.7 ]
+        torch-version: [ 1.5.0]
+        tensorflow-version: [ 1.15.0 ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Cache pip
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('setup.py') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+            ${{ runner.os }}-
+      - name: Install dependencies
+        run: |
+          python -m pip install --progress-bar off --upgrade pip
+          pip install --progress-bar off pytest==5.1.3
+          pip install --progress-bar off coverage ddt
+      - name: Install the project
+        run: |
+          pip install .[allennlp]
+      - name: Test with pytest and run coverage
+        run: |
+          coverage run -m pytest tests/allennlp_processors_test.py
```

Comment:
I think there are two lines different in most tasks, should add them in the matrix and only use variables to avoid repeating the bolierplate

Reply:
for the matrix, we should be able to do something like:
`matrix:
        python-version: [ 3.6, 3.7 ]
        torch-version: [ 1.5.0]
        tensorflow-version: [ 1.15.0 ]
        test-details: [
            [dep: "allennlp", test_file: "tests/allennlp_processors_test.py"], 
            [dep: "varder, nltk", test_file: "tests/nltk_processors_test.py" ]
        ]
`

And for the two specific lines, we do:
`pip install .[${{ matrix.test-details.test_file }}]`
`coverage run -m pytest ${{ matrix.test-details.dep }}`

Reference: https://stackoverflow.com/questions/66025220/paired-values-in-github-actions-matrix